### DISCUSSION TITLE
Overriding remote_host with "X-Forwarded-For" header - Access

### DISCUSSION THREAD

**Discussion by WesPr** [logfellow/logstash-logback-encoder#820] [Q&A] [answered]
Hi, 
Is there perhaps a way to override the default remoteHost access field with the "X-Forwarded-For" header in logstash access logs? I currently have my application logging the AWS load balancer IP and not the client-IP. 
Thank you.
Kind regards, 
Wesley

**Comment by brenuart** [comment, marked as answer]
Hi Wes,

I'm sorry but there is no such option at the moment but you can always implement a custom JsonProvider to process the request headers yourself. You can start with `RemoteHostJsonProvider` as an example (no need to implement the `FieldNamesAware` interface). See https://github.com/logfellow/logstash-logback-encoder#custom-json-provider for more information.

Hope this helps.
Do not hesitate to come back here if you need more information.

**Reply by WesPr** [reply]
Hi Bertrand,

Thank you so much for your answer, I will implement a custom  JsonProvider. 

Have a great day!
